packages/server-wallet/src/models/ledger-request.ts

```diff
@@ -0,0 +1,106 @@
+import {Model, TransactionOrKnex} from 'objection';
+
+import {Bytes32} from '../type-aliases';
+
+export type LedgerRequestStatus =
+  | 'pending' // Request added to DB to be approved or rejected by queue
+  // | 'rejected' // Rejected due to lack of available funds TODO: Implement
+  | 'succeeded' // Ledger update became supported and thus request succeeded
+  | 'failed'; // Rejected for an unexpected reason (some error occurred) TODO: Implement
+
+export interface LedgerRequestType {
+  ledgerChannelId: Bytes32;
+  channelToBeFunded: Bytes32;
+  status: LedgerRequestStatus;
+  type: 'fund' | 'defund';
+}
+
+export class LedgerRequest extends Model implements LedgerRequestType {
+  channelToBeFunded!: LedgerRequestType['channelToBeFunded'];
+  ledgerChannelId!: LedgerRequestType['ledgerChannelId'];
+  status!: LedgerRequestType['status'];
+  type!: LedgerRequestType['type'];
+
+  static tableName = 'ledger_requests';
+  static get idColumn(): string[] {
+    return ['channelToBeFunded', 'type'];
+  }
+
+  static async getRequest(
+    channelToBeFunded: Bytes32,
+    type: 'fund' | 'defund',
+    tx: TransactionOrKnex
+  ): Promise<LedgerRequestType> {
```

Comment:
Should this be `Promise<LedgerRequestType | undefined>`?

Reply:
@tomclose I think you could ask this question of many of our models/methods. 

For example: 
https://github.com/statechannels/statechannels/blob/ecb248c2cbfd1336f93cdbe5e39b43414729b0ba/packages/server-wallet/src/models/channel.ts#L119-L125